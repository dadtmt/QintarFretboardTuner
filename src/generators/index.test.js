import {addSemitone, areInSameOctave, generateFret, generateString} from './index'

describe ("areInSameOctave", () => {
    it("E4 and F4 are in the same octave", () => {
        expect(areInSameOctave("e4", "f4")).toBeTruthy()
    })
    it("F4 and E5 are same octave", () => {
        expect(areInSameOctave("f4", "e5")).toBeTruthy()
    })
    it("E4 and E5 are not in the same octave", () => {
        expect(areInSameOctave("e4", "e5")).toBeFalsy()
    })
    it("E4 and F5 are not the same octave", () => {
        expect(areInSameOctave("e4", "f5")).toBeFalsy()
    })
})

describe ("generateFret", () => {
    it("returns next semitone", () => {
        expect(generateFret('E4')('E4')).toEqual(['F4','F4'])
    })
    it("returns next semitone", () => {
        expect(generateFret('E4')('F4')).toEqual(['Gb4', 'Gb4'])
    })
    it("returns false when fret is the octave above the flat note", () => {
        expect(generateFret('E4')('E5')).toBeFalsy()
    })
})

it('generates E4 string', () => {
    expect(generateString('e4')).toMatchSnapshot()
})
