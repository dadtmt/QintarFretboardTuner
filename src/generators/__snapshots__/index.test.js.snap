// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates A3 string 1`] = `
Array [
  "A#3",
  "B3",
  "C4",
  "Db4",
  "D4",
  "D#4",
  "E4",
  "F4",
  "Gb4",
  "G4",
  "Ab4",
  "A4",
]
`;

exports[`generates E4 string 1`] = `
Array [
  "F4",
  "Gb4",
  "G4",
  "Ab4",
  "A4",
  "A#4",
  "B4",
  "C5",
  "Db5",
  "D5",
  "Eb5",
  "E5",
]
`;

exports[`generates a 10 P5 StringsFlatNote 1`] = `
Array [
  "G#6",
  "C#6",
  "F#5",
  "B4",
  "E4",
  "A3",
  "D3",
  "G2",
  "C2",
  "F1",
]
`;

exports[`generates a P5 StringsFlatNote 1`] = `
Array [
  "E4",
  "A3",
  "D3",
  "G2",
  "C2",
  "F1",
]
`;

exports[`generates a Quintar fretboard 1`] = `
Array [
  Object {
    "flatNote": "E4",
    "frets": Array [
      "F4",
      "Gb4",
      "G4",
      "Ab4",
      "A4",
      "A#4",
      "B4",
      "C5",
      "Db5",
      "D5",
      "Eb5",
      "E5",
    ],
  },
  Object {
    "flatNote": "A3",
    "frets": Array [
      "A#3",
      "B3",
      "C4",
      "Db4",
      "D4",
      "D#4",
      "E4",
      "F4",
      "Gb4",
      "G4",
      "Ab4",
      "A4",
    ],
  },
  Object {
    "flatNote": "D3",
    "frets": Array [
      "D#3",
      "E3",
      "F3",
      "Gb3",
      "G3",
      "G#3",
      "A3",
      "A#3",
      "B3",
      "C4",
      "Db4",
      "D4",
    ],
  },
  Object {
    "flatNote": "G2",
    "frets": Array [
      "G#2",
      "A2",
      "A#2",
      "B2",
      "C3",
      "C#3",
      "D3",
      "D#3",
      "E3",
      "F3",
      "Gb3",
      "G3",
    ],
  },
  Object {
    "flatNote": "C2",
    "frets": Array [
      "C#2",
      "D2",
      "D#2",
      "E2",
      "F2",
      "F#2",
      "G2",
      "G#2",
      "A2",
      "A#2",
      "B2",
      "C3",
    ],
  },
  Object {
    "flatNote": "F1",
    "frets": Array [
      "F#1",
      "G1",
      "G#1",
      "A1",
      "A#1",
      "B1",
      "C2",
      "C#2",
      "D2",
      "D#2",
      "E2",
      "F2",
    ],
  },
]
`;
